mixer/adapter/rbac/rbacStore.go

```diff
@@ -127,6 +127,126 @@ func (rs *RolesMapByNamespace) AddServiceRoleBinding(name, namespace string, pro
 	return nil
 }
 
+// ListPermissions returns list of AccessRule(permission) that matches subjects.
+func (rs *ConfigStore) ListPermissions(subject SubjectArgs, action ActionArgs) (*map[string]*rbacproto.ServiceRole, error) {
+	instance, err := createInstance(subject, action)
+	if err != nil {
+		return nil, err
+	}
+
+	logger := istioctlLogger{}
+
+	namespace := instance.Action.Namespace
+	if namespace == "" {
+		return nil, logger.Errorf("Missing namespace")
+	}
+
+	accessRules := make(map[string]*rbacproto.ServiceRole)
+
+	rn := rs.Roles[namespace]
+	if rn == nil {
+		return &accessRules, nil
+	}
+
+	for roleName, roleInfo := range rn {
+		logger.Infof("Checking role: %s", roleName)
+
+		for _, serviceRoleBinding := range roleInfo.Bindings {
+			for _, subject := range serviceRoleBinding.Subjects {
+				if checkSubjectsMatch(subject, instance) {
+					accessRules[roleName] = rn[roleName].Info
+					break
+				}
+			}
+
+			if _, ok := accessRules[roleName]; ok {
+				break
+			}
+		}
+	}
+
+	return &accessRules, nil
+}
+
+// ListMembers searches users that allowed to use the given service role
+func (rs *ConfigStore) ListMembers(subject SubjectArgs, action ActionArgs) (*[]string, error) {
+	instance, err := createInstance(subject, action)
+	if err != nil {
+		return nil, err
+	}
+
+	logger := istioctlLogger{}
+
+	namespace := instance.Action.Namespace
+	if namespace == "" {
+		return nil, logger.Errorf("Missing namespace")
+	}
+
+	serviceName := instance.Action.Service
+	if serviceName == "" {
+		return nil, logger.Errorf("Missing service")
+	}
+
+	path := instance.Action.Path
+	if path == "" {
+		return nil, logger.Errorf("Missing path")
+	}
+
+	method := instance.Action.Method
+	if method == "" {
+		return nil, logger.Errorf("Missing method")
+	}
+
+	membersDedup := make(map[string]bool)
+	members := make([]string, 0)
+
+	rn := rs.Roles[namespace]
+	if rn == nil {
+		return &members, nil
+	}
+
+	for roleName, roleInfo := range rn {
+		eligibleRole := false
+
+		logger.Infof("Checking role: %s", roleName)
+		rules := roleInfo.Info.GetRules()
+		for _, rule := range rules {
+			if matchRule(serviceName, path, method, instance.Action.Properties, rule, logger) {
+				eligibleRole = true
+				logger.Debugf("rule matched")
+				break
+			}
+		}
+
+		if !eligibleRole {
+			logger.Infof("role %s is not eligible", roleName)
+			continue
+		}
+
+		for _, serviceRoleBinding := range roleInfo.Bindings {
+			for _, subject := range serviceRoleBinding.Subjects {
```

Comment:
Yeah, I think it makes more sense to also print groups and properties.
`user` is just one field in the ServiceRoleBinding, customer could choose to use it or not, or together with other fields, like `group` and `property`.

Reply:
Changed "members" to "subjects". 

```
$istioctl experimental rbac  list subjects GET reviews.default.svc.cluster.local / -a destination.labels[app]=productpage 
                        USER                         GROUP             PROPERTIES            
 -------------------------------------------------- ------- -------------------------------- 
  cluster.local/ns/default/sa/bookinfo-productpage                                           
                                                             source.namespace: istio-system  
                                                             source.namespace: default     

$ istioctl experimental rbac list subjects GET reviews.default.svc.cluster.local / -a destination.labels[app]=productpage -o yaml
- user: cluster.local/ns/default/sa/bookinfo-productpage
- properties:
    source.namespace: istio-system
- properties:
    source.namespace: default
```